File: StudentManagement/src/main/java/com/Student/StudentManagement/Service/StudentService.java
package com.Student.StudentManagement.Service;


import com.Student.StudentManagement.Entity.Student;
import com.Student.StudentManagement.Repository.StudentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class StudentService {

    @Autowired
    private StudentRepository studentRepository;

    // Get all students
    public List<Student> getAllStudents() {
        return studentRepository.findAll();
    }

    // Get a student by ID
    public Optional<Student> getStudentById(Long id) {
        return studentRepository.findById(id);
    }

    // Add a student
    public Student addStudent(Student student) {
        return studentRepository.save(student);
    }

    // Update a student
    public Student updateStudent(Long id, Student student) {
        Optional<Student> existingStudent = studentRepository.findById(id);
        if (existingStudent.isPresent()) {
            Student updatedStudent = existingStudent.get();
            updatedStudent.setFirstName(student.getFirstName());
            updatedStudent.setLastName(student.getLastName());
            updatedStudent.setEmail(student.getEmail());
            updatedStudent.setPhno(student.getPhno());
            updatedStudent.setBranch(student.getBranch());
            return studentRepository.save(updatedStudent);
        } else {
            return null;
        }
    }

    // Delete a student
    public void deleteStudent(Long id) {
        studentRepository.deleteById(id);
    }
}

File: StudentManagement-frontend/student-management/src/StudentService.js
// src/StudentService.js
import axios from 'axios';

// const API_URL = 'http://localhost:8080/students';
const API_URL = 'http://localhost:5000/students';

export const getStudents = async () => {
    const response = await axios.get(API_URL);
    return response.data;
};

export const getStudentById = async (id) => {
    const response = await axios.get(`${API_URL}/${id}`);
    return response.data;
};

export const addStudent = async (student) => {
    const response = await axios.post(API_URL, student);
    return response.data;
};

export const updateStudent = async (id, student) => {
    const response = await axios.put(`${API_URL}/${id}`, student);
    return response.data;
};

export const deleteStudent = async (id) => {
    await axios.delete(`${API_URL}/${id}`);
};